featuretools/primitives/utils.py

```diff
@@ -17,28 +18,47 @@
 from featuretools.utils.gen_utils import Library, find_descendents
 
 
-# returns all aggregation primitives, regardless of compatibility
-def get_aggregation_primitives():
-    aggregation_primitives = set([])
+def _get_primitives(primitive_kind):
+    """Helper function that selects all primitives
+    that are instances of `primitive_kind`
+    """
+    primitives = set()
     for attribute_string in dir(featuretools.primitives):
         attribute = getattr(featuretools.primitives, attribute_string)
         if isclass(attribute):
-            if issubclass(attribute, featuretools.primitives.AggregationPrimitive):
-                if attribute.name:
-                    aggregation_primitives.add(attribute)
-    return {prim.name.lower(): prim for prim in aggregation_primitives}
+            if issubclass(attribute, primitive_kind) and attribute.name:
+                primitives.add(attribute)
+    return {prim.name.lower(): prim for prim in primitives}
+
+
+def get_aggregation_primitives():
+    """Returns all aggregation primitives, regardless
+    of compatibility
+    """
+    return _get_primitives(featuretools.primitives.AggregationPrimitive)
 
 
-# returns all transform primitives, regardless of compatibility
 def get_transform_primitives():
-    transform_primitives = set([])
-    for attribute_string in dir(featuretools.primitives):
-        attribute = getattr(featuretools.primitives, attribute_string)
-        if isclass(attribute):
-            if issubclass(attribute, featuretools.primitives.TransformPrimitive):
-                if attribute.name:
-                    transform_primitives.add(attribute)
-    return {prim.name.lower(): prim for prim in transform_primitives}
+    """Returns all transform primitives, regardless
+    of compatibility
+    """
+    return _get_primitives(featuretools.primitives.TransformPrimitive)
+
+
+def get_natural_language_primitives():
```

Comment:
I'm wondering about the need to create this new public method? If the only reason we are doing this is for running a single test, maybe this should just be part of the test, or perhaps a specific test fixture instead?

Reply:
I can see it being useful if someone wanted to feed all NL primitives to DFS.

Reply:
Sure, but that seems beyond the scope of this MR.

My current thoughts:
1. Let's not complicate our API with new methods unless we have a specific need for the functionality
2. If we do add this method for the purpose above, we should update our API docs to show we added this.
3. If we add a method like this, should it be specific to NaturalLanguage or should it be something more generic? Should it match on only logical types or should it match on `ColumnSchema` which really are what fully define the type? I think we should answer those questions before we implement a new public method.